libfreerdp/core/connection.c

```diff
@@ -661,7 +661,8 @@ BOOL rdp_client_redirect(rdpRdp* rdp)
 			return FALSE;
 	}
 
-	settings->RdstlsSecurity = settings->RedirectionFlags & LB_PASSWORD_IS_PK_ENCRYPTED;
+	settings->RdstlsSecurity =
+	    (settings->RedirectionFlags & LB_PASSWORD_IS_PK_ENCRYPTED) != 0 ? TRUE : FALSE;
```

Comment:
Could also be `!!(settings->RedirectionFlags & LB_PASSWORD_IS_PK_ENCRYPTED)` like they do in `glib` and related projects...